src/components/CytoscapeGraph/ContextMenu/NodeContextMenu.tsx

```diff
@@ -1,49 +1,109 @@
 import * as React from 'react';
 import { NodeContextMenuProps } from '../CytoscapeContextMenu';
+import { JaegerSearchOptions, JaegerURLSearch } from '../../JaegerIntegration/RouteHelper';
 import history from '../../../app/History';
+import { Paths } from '../../../config';
 
-export class NodeContextMenu extends React.PureComponent<NodeContextMenuProps> {
+type NodeContextMenuState = {
+  nodeType: string;
+  app: string | undefined;
+};
+
+export class NodeContextMenu extends React.PureComponent<NodeContextMenuProps, NodeContextMenuState> {
+  constructor(props: NodeContextMenuProps) {
+    super(props);
+    let app: string | undefined = '';
+    let nodeType = '';
+    switch (this.props.nodeType) {
+      case 'app':
+        nodeType = Paths.APPLICATIONS;
+        app = this.props.app;
+        break;
+      case 'service':
+        nodeType = Paths.SERVICES;
+        app = this.props.service;
+        break;
+      case 'workload':
+        app = this.props.workload;
+        nodeType = Paths.WORKLOADS;
+        break;
+      default:
+    }
+    this.state = { nodeType, app };
```

Comment:
I think that we need to use the componentDidUpdate to update this state, it could bite us later if this instance is re-used by redux when building the menu instead of creating it from scratch.